src/egl/drivers/dri2/platform_android.c

```diff
@@ -657,28 +658,44 @@ droid_create_image_from_prime_fd_yuv(_EGLDisplay *disp, _EGLContext *ctx,
 {
    struct dri2_egl_display *dri2_dpy = dri2_egl_display(disp);
    struct android_ycbcr ycbcr;
+   gralloc1_rect_t accessRegion;
    size_t offsets[3];
    size_t pitches[3];
    int is_ycrcb;
    int fourcc;
    int ret;
-   return 0;
-#ifdef TENTH
-   if (!dri2_dpy->gralloc->lock_ycbcr) {
-      _eglLog(_EGL_WARNING, "Gralloc does not support lock_ycbcr");
-      return NULL;
-   }
+   if(dri2_dpy->gralloc_version == HARDWARE_MODULE_API_VERSION(1, 0)) {
+     if (!dri2_dpy->pfn_lockflex) {
+        _eglLog(_EGL_WARNING, "Gralloc does not support lockflex");
+        return NULL;
+     }
+
+     memset(&ycbcr, 0, sizeof(ycbcr));
+     ret = dri2_dpy->pfn_lockflex(dri2_dpy->gralloc1_dvc, buf->handle,
+                                       0, 0, &accessRegion, &ycbcr, 0);
```

Comment:
cannot pass android_ycbcr struct to lockflex function, it takes different type android_flex_layout, so we will need to abstract this part of code, maybe as a separate function that fetches the information and fills pitches, offsets etc.